event_scheduler/event_scheduler.py

```diff
@@ -257,7 +257,7 @@ def cancel(self, event: Event) -> int:
             if self._scheduler_status != SchedulerStatus.RUNNING:
                 return -1
             try:
-                if self._queue[0] == event:
+                if self._queue and self._queue[0] == event:
                     self._notify()
                 self._queue.remove(event)
```

Comment:
if self._queue is None won't that be problematic with line 262? im not sure if its silly to make that its own check above this if statement and do continue if its true

Reply:
nvm ignore this comment i see its always just gonna be an empty array if anything